src/protosym/simplecas.py

```diff
@@ -353,37 +356,24 @@ def from_sympy(expr: Any) -> Expr:
 eval_latex.add_atom(Integer.atom_type, str)
 eval_latex.add_atom(Symbol.atom_type, str)
 eval_latex.add_atom(Function.atom_type, str)
+eval_latex.add_op_generic(lambda head, args: f'{head}({", ".join(args)})')
 eval_latex.add_op1(sin.rep, lambda a: rf"\sin({a})")
 eval_latex.add_op1(cos.rep, lambda a: rf"\cos({a})")
 eval_latex.add_op2(Pow.rep, lambda b, e: f"{b}^{{{e}}}")
 eval_latex.add_opn(Add.rep, lambda args: f'({" + ".join(args)})')
 eval_latex.add_opn(Mul.rep, lambda args: "(%s)" % r" \times ".join(args))
 
-_bin_expand = Transformer()
-_bin_expand.add_opn(Add.rep, lambda args: reduce(Add.rep, args))
-_bin_expand.add_opn(Mul.rep, lambda args: reduce(Mul.rep, args))
-
-
-def _op1(a: int | str) -> int:
-    return 1
-
-
-def _sum1(*a: int) -> int:
-    return 1 + sum(a)
-
-
-def _sum1n(a: Iterable[int]) -> int:
-    return 1 + sum(a)
-
+bin_expand = Transformer()
+bin_expand.add_opn(Add.rep, lambda args: reduce(Add.rep, args))
+bin_expand.add_opn(Mul.rep, lambda args: reduce(Mul.rep, args))
 
+#
+# Maybe it should be possible to just pass these as arguments to the Evaluator
+# constructor.
+#
```

Comment:
I was wondering the same thing when looking at `Evaluator` above. However, for the time being I think the way that it is very clearly demonstrates how it operates without complicating the class with multiple ways to achieve the same outcome.

Reply:
Yes, and later I would expect the interface for specifying these rules to be different anyway so there will already be a layer I think on top of what is there.